base/iterators.jl

```diff
@@ -1341,20 +1341,29 @@ mutable struct Stateful{T, VS}
     itr::T
     # A bit awkward right now, but adapted to the new iteration protocol
     nextvalstate::Union{VS, Nothing}
-    taken::Int
+    # Number of remaining elements, or nothing if not known or infinite
+    remaining::Union{Int, Nothing}
```

Comment:
```suggestion
    # Number of remaining elements, or nothing if not known or infinite
    remaining::N
```
(making this a new type parameter)